### ISSUE TITLE
rst files are not tested for astropy-helpers versions between v1.0 >= v.0.4.4

### ISSUE THREAD

**Issue by bsipocz** [astropy/astropy-helpers#143]
``rst`` files are not tested on the v0.4.x branch since v0.4.4 (see https://github.com/astropy/photutils/pull/251#issuecomment-75400091). I didn't find what went wrong, but v.1.0 is fine. 

Sadly there wasn't an error raised, and the rst files were silently ignored (on travis, too). Directly testing a single ``rst`` file with the ``-t`` option, I got the following error:

```
$  python setup.py test -t docs/photutils/detection.rst
running test
running build
running build_py
running build_ext
skipping 'photutils/geometry/circular_overlap.c' Cython extension (up-to-date)
skipping 'photutils/geometry/core.c' Cython extension (up-to-date)
skipping 'photutils/geometry/elliptical_overlap.c' Cython extension (up-to-date)
skipping 'photutils/geometry/rectangular_overlap.c' Cython extension (up-to-date)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "photutils/_astropy_init.py", line 113, in test
    coverage=coverage, open_files=open_files, **kwargs)
  File "/sw/lib/python2.7/site-packages/astropy/tests/helper.py", line 153, in run_tests
    "Can not test .rst files without a docs_path "
ValueError: Can not test .rst files without a docs_path specified.
```

Due to this bug, the rst doctests didn't run for a month for ``photutils``, and this may affect other affiliated packages that haven't upgraded to v.1.0 yet.

**Comment by embray** [comment]
That sucks because [the change](https://github.com/astropy/astropy-helpers/commit/1068e261d292814156e6c21f4f1f0a97dcffacc3) in astropy-helpers that might have caused this was intended exactly to avoid problems like this.  I think the problem is that the package-template was already mitigating this problem by another means here:

https://github.com/astropy/photutils/blob/master/photutils/_astropy_init.py#L101

And the the two approaches didn't play well with each other, since by using `**kwargs` in the `test()` function for affiliated packages the test runner wasn't picking up that it supported certain arguments (namely `docs_path`).

I still need to make some further improvements on the astropy end so that there aren't three separate interfaces to the test runner that have to be maintained.  This change was part of that goal, but there are still at least two interfaces, and really should just be one.

**Comment by embray** [comment]
It seems the only reason this was "fixed" in 1.0 is that I never copied that change over to the master branch, so in fact the fact that 1.0 fixes it is a bug.  This should be changed in the package-template and the change should be better communicated.

**Comment by embray** [comment]
This should be a moot point since #184 .  I think this may still be broken on the 0.4.x branch, but I don't think there will be any more releases for that, and all the affiliated packages should have been updated by now.
